docs/guides/artifacts/understanding-artifact-definitions.md

```diff
@@ -0,0 +1,148 @@
+title: Understanding Artifact Definitions
+sidebar_label: Understanding Artifact Definitions
+---
+
+# Understanding Artifact Definitions
+
+Artifact definitions are JSON Schema-based contracts that define how infrastructure components can interact with each other in Massdriver. They serve as the foundation for connecting different infrastructure components and enabling seamless integration between them.
+
+## Overview
+
+Artifact definitions serve several key purposes:
+
+1. **Contract Definition**: They define the contract between Infrastructure as Code (IaC) module executions/runs
+2. **State Transit**: They enable state to be passed between different IaC tools
+3. **Platform Extension**: They extend cloud support and UI capabilities of the platform
+4. **Serialization**: They define how artifacts can be downloaded and serialized
+
+## Structure
+
+Every artifact definition must contain two top-level fields:
+
+1. `data` (object): Contains the encrypted data that will be stored at rest
+2. `specs` (object): Contains UI integrations and display data
+
+### Example Structure
+
+```json
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$id": "https://api.massdriver.cloud/artifact-definitions/ORG/NAME",
+  "type": "object",
+  "required": ["data", "specs"],
+  "properties": {
+    "data": {
+      "type": "object",
+      "properties": {
+        // Data properties here
+      }
+    },
+    "specs": {
+      "type": "object",
+      "properties": {
+        // Spec properties here
+      }
+    }
+  }
+}
+```
+
+## Identification
+
+Artifact definitions are identified using the following pattern:
+
+```
+ORG_SLUG/ARTIFACT_DEFINITION_TYPE_NAME
+```
+
+For example: `massdriver/aws-s3-bucket`
+
+The pattern follows this regex:
+```
+[a-z0-9-]+\/[a-z0-9-]+
+```
+
+## Usage in massdriver.yaml
+
+Artifact definitions are referenced in massdriver.yaml files under two main sections:
+
+1. `:artifacts`: Defines the artifacts that a bundle can produce
+2. `:connections`: Defines the artifacts that a bundle can consume
+
+An example `massdriver.yaml` file for an RDS OpenTofu Module:
+```yaml
+# The database will require a network artifact to put the database in.
+connections:
+  required:
+  - vpc
+  properties:
+    vpc:
+      $ref: massdriver/aws-vpc
+
+# The will emit an artifact with security group, db connection secrets, and iam policies
+artifacts:
+  required:
+  - database
+  properties:
+    database:
+      $ref: massdriver/aws-rds-postgres
+```
+
+For more information about bundle configuration, see [Bundle Configuration](#TODO).
+
+## Artifact Lifecycle and Connection Phases
+
+The lifecycle of artifact connections spans from initial package linking to final data injection. Through distinct phases of type validation and data exchange, artifact definitions ensure type safety and data integrity across your deployment pipeline.
+
+### Nominal Typing (Simple Explanation)
```

Comment:
Do you intend to leave this with both the technical and simple explanations in the final document?